tests: disable ML-DSA tesing in kryoptic on Fedora < 42

The-Mule opened the pull request:
#### Description

Kryoptic 1.2 is needed for ML-DSA tests and it is not available in Fedora until version 43 (FYI, kryoptic token is currently only available on Fedora).

#### Checklist

N/A

#### Reviewer's checklist:

- [ ] Any issues marked for closing are addressed
- [ ] There is a test suite reasonably covering new functionality or modifications
- [ ] This feature/change has adequate documentation added
- [ ] Code conform to coding style that today cannot yet be enforced via the check style test
- [ ] Commits have short titles and sensible commit messages
- [ ] Coverity Scan has run if needed (code PR) and no new defects were found

simo5 commented on tests/kryoptic-init.sh:
```diff
@@ -51,4 +51,13 @@ export TOKENCONFIGVARS="export KRYOPTIC_CONF=$TOKDIR/kryoptic.conf"
+
+    # ML-DSA support is available since version 1.2
+    SUPPORT_ML_DSA=0
+    if rpm -q kryoptic &>/dev/null && [[ "$(rpm -q --qf '%{V}' kryoptic)" =~ 1\.[01]\. ]]; then
```
Sorry I was not clear, but I do not want this check in this file, as it will cause unnecessary errors on non RPM systems.

This file should only support the ability to override the  SUPPORT_ML_DSA from an external job and default to 1 otherwise.

The CI jobs can then test kryoptic's version based on whatever packaging system the build is running on and set the override there.

simo5 reviewed (approved):
LGTM

The-Mule commented:
Added forgotten export to setup.sh (`export SUPPORT_ED448="${SUPPORT_ED448}"`) and replaced `test -z` with `if [ -z` in kryoptic-init.sh (to avoid setup fail when SUPPORT_ML_DSA is already defined). Verified carefully that the mldsa test runs with kryoptic and that it can be disabled by setting `SUPPORT_ML_DSA=0`.

Now it's finally good to go.
